x/pricefeed/simulation/operations/msg.go

```diff
@@ -0,0 +1,157 @@
+		// (1) RANDOMLY PICK AN ASSET OUT OF BNB AN BTC [TODO QUESTION - PRESUMABLY USDX IS EXCLUDED AS IT IS A STABLE DENOM?]
+		// (2) GET THE CURRENT PRICE OF THAT ASSET IN USD
+		// (3) GENERATE A RANDOM NUMBER IN THE RANGE 0.8-1.2 (UNIFORM DISTRIBUTION) [TODO POTENTIAL UPDATE MAKE IT NORMALLY DISTRIBUTED ACROSS A WIDER RANGE?]
+		// (4) MULTIPLY THE CURRENT PRICE BY THE RANDOM NUMBER
+		// (5) POST THE NEW PRICE TO THE KEEPER
+
+		// pick a random asset out of BNB and BTC
+		assetCode := pickRandomAsset(ctx, keeper, r)
+
+		fmt.Printf("Picked asset: %s\n", assetCode)
+
+		//GET THE CURRENT PRICE OF THE ASSET
+		marketID := strings.ToLower(assetCode + ":usd") // convert to lower case
+
+		currentPrice, err := keeper.GetCurrentPrice(ctx, marketID) // TODO NOTE THIS IS marketID AND **NOT** THE assetcode
+		if err != nil {
+			return noOpMsg, nil, fmt.Errorf("Error getting current price")
+		}
+
+		fmt.Println("Got oracles:")
+		oracles, err := keeper.GetOracles(ctx, marketID)
+		if err != nil {
+			return noOpMsg, nil, fmt.Errorf("Error getting oracles")
+		}
+		address := oracles[0]
+		// fmt.Print(oracles[0])
+		// fmt.Println()
+		// fmt.Print(oracles[0].String)
+		// fmt.Println()
+
+		// get the address for the account
+		// this address needs to be an oracle and also exist. genesis should add all the accounts as oracles.
+		// address := sdk.AccAddress(accs[0].Address)
+
+		// generate a new random price based off the current price
+		price, err := pickNewRandomPrice(r, currentPrice.Price)
+		if err != nil {
+			return noOpMsg, nil, fmt.Errorf("Error picking random price")
+		}
+
+		// get the expiry time based off the current time
+		expiry := getExpiryTime(ctx)
+
+		// MSG POST PRICE
+		// GENERATE THE MSG TO SEND TO THE KEEPER
+		// now create the msg to post price
+		msg := types.NewMsgPostPrice(address, marketID, price, expiry)
+
+		// Perform basic validation of the msg - don't submit errors that fail ValidateBasic, use unit tests for testing ValidateBasic
+		if err := msg.ValidateBasic(); err != nil {
+			return noOpMsg, nil, fmt.Errorf("expected msg to pass ValidateBasic: %s", msg.GetSignBytes())
+		}
+
+		fmt.Printf("Trying to submit msg: %s\n", msg)
+
+		// now we submit the pricefeed update message
+		// TODO QUESTION - this is failing for some reason? Any ideas why?
+		if ok := submitMsg(ctx, handler, msg); !ok {
+			return noOpMsg, nil, fmt.Errorf("could not submit pricefeed msg")
+		}
+		fmt.Println("Pricefeed update sumbitted!")
+		return simulation.NewOperationMsg(msg, true, "pricefeed update submitted"), nil, nil // TODO what should go in comment field?
+	}
+}
+
+// pickRandomAsset picks a random asset out of the assets with equal probability
+func pickRandomAsset(ctx sdk.Context, keeper keeper.Keeper, r *rand.Rand) (asset string) {
+	// get the params
+	params := keeper.GetParams(ctx)
+	// now pick a random asset
+	randomIndex := simulation.RandIntBetween(r, 0, len(params.Markets))
+	randomAsset := params.Markets[randomIndex]
```

Comment:
Return the entire `Market` rather than the base asset (we can then access market.MarketID instead of computing it manually).

Reply:
🆗